core/iwasm/interpreter/wasm_loader.c

```diff
@@ -7129,6 +7129,40 @@ check_branch_block(WASMLoaderContext *loader_ctx, uint8 **p_buf, uint8 *buf_end,
     return NULL;
 }
 
+#if WASM_ENABLE_EXCE_HANDLING != 0
+static BranchBlock *
+check_branch_block_for_delegate(WASMLoaderContext *loader_ctx, uint8 **p_buf,
+                                uint8 *buf_end, char *error_buf,
+                                uint32 error_buf_size)
+{
+    uint8 *p = *p_buf, *p_end = buf_end;
+    BranchBlock *frame_csp_tmp;
+    uint32 depth;
+
+    read_leb_uint32(p, p_end, depth);
+    /*
+     * Note: "delegate 0" means the surrounding block, not the
+     * try-delegate block itself.
+     *
+     * Note: the caller hasn't popped the try-delegate frame yet.
+     */
+    bh_assert(loader_ctx->csp_num > 0);
+    if (loader_ctx->csp_num - 1 <= depth) {
```

Comment:
Not sure whether the check is enough? The original code calls check_branch_block:
check_branch_block -> CHECK_BR -> wasm_loader_check_br
And in wasm_loader_check_br, there are more checks:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/3111a86cb341cee9eb0b3e7d8d1d43327846a627/core/iwasm/interpreter/wasm_loader.c#L7058-L7088

Is it better to use `CHECK_BR(depth + 1)`?

Reply:
the most of checks in wasm_loader_check_br are not appropriate for delegate, where parameters are given via the exception, not stack.

Reply:
OK, got it, thanks!